src/compiler/msvc.rs

```diff
@@ -340,13 +350,15 @@ pub fn preprocess<T>(creator: &T,
 
     let parsed_args = parsed_args.clone();
     let includes_prefix = includes_prefix.to_string();
-    let cwd = cwd.to_string();
+    let cwd = cwd.to_owned();
 
     Box::new(run_input_output(cmd, None).and_then(move |output| {
         let parsed_args = &parsed_args;
         if let (Some(ref objfile), &Some(ref depfile)) = (parsed_args.outputs.get("obj"), &parsed_args.depfile) {
-            let mut f = File::create(Path::new(&cwd).join(depfile))?;
-            write!(f, "{}: {} ", objfile, parsed_args.input)?;
+            let mut f = File::create(cwd.join(depfile))?;
+
+            // TODO: these calls to `display` are lossy ...
```

Comment:
This isn't really worse than what we had before, where a source file name that wasn't valid Unicode would just fail somewhere along the way, but it's a little worrying that we could be writing bad data here and silently breaking things. In any event, you don't have to fix this right now. I was going to say you could use [`local_encoding`](https://docs.rs/local-encoding/0.2.0/local_encoding/) like we're using to decode the MSVC output, but that only deals with turning strings into bytes and that doesn't actually help. I guess we could open an issue on `local_encoding` to support encoding `&[u16]` or some such to bytes, and then pass it `input.to_wide()`? It'd basically just need to call `WideCharToMultiByte`, and I'm sure it's already doing something very similar.